pyogrio/tests/test_geopandas_io.py

```diff
@@ -460,6 +460,51 @@ def test_write_dataframe(tmp_path, naturalearth_lowres, ext):
     )
 
 
+@pytest.mark.parametrize("ext", [ext for ext in ALL_EXTS if ext != ".fgb"])
+def test_write_dataframe_nogeom(tmp_path, naturalearth_lowres, ext):
+    """Test writing a dataframe, so without a geometry column.
+
+    FlatGeobuf (.fgb) doesn't seem to support this, and just writes an empty file.
+    """
+    # Prepare test data
+    input_gdf = read_dataframe(naturalearth_lowres)
+    input_df = input_gdf.drop(columns=["geometry"])
+    assert isinstance(input_df, pd.DataFrame)
+    assert not isinstance(input_df, gp.GeoDataFrame)
+
+    output_path = tmp_path / f"test{ext}"
+    write_dataframe(input_df, output_path)
+
+    if ext == ".shp":
+        # A shapefile without geometry column results in only a .dbf file.
+        output_path = output_path.with_suffix(".dbf")
+    assert output_path.exists()
+    result_df = read_dataframe(output_path)
+
+    assert isinstance(result_df, pd.DataFrame)
+
+    # some dtypes do not round-trip precisely through these file types
+    check_dtype = ext not in [".json", ".geojson", ".geojsonl", ".xlsx"]
```

Comment:
Not really to ALL_EXT but specifically to this test because xlsx is now removed from the unsupported drivers.

But it kind of depends on the extended write driver detection so I'll remove it... xlsx isn't that critical anyway I suppose.